k256/src/arithmetic/mul.rs

```diff
@@ -339,6 +339,40 @@ impl LinearCombination<[(ProjectivePoint, Scalar)]> for ProjectivePoint {
     }
 }
 
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+use risc0_bigint2::ec;
+
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+use super::{affine_to_projective, projective_to_affine, scalar_to_words};
+
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+fn lincomb(
+    xks: &[(ProjectivePoint, Scalar)],
+    tables: &mut [(LookupTable, LookupTable)],
+    digits: &mut [(Radix16Decomposition<33>, Radix16Decomposition<33>)],
+) -> ProjectivePoint {
+    let mut xks_iter = xks
+        .iter()
+        .map(|(p, s)| (projective_to_affine(p), scalar_to_words(s)));
+    let Some((affine, scalar)) = xks_iter.next() else {
+        return ProjectivePoint::IDENTITY;
+    };
+
+    let mut result = ec::AffinePoint::new_unchecked([0u32; 8], [0u32; 8]);
+    affine.mul(&scalar, &mut result);
+    let mut result_buffer = ec::AffinePoint::new_unchecked([0u32; 8], [0u32; 8]);
+    let mut mul_buffer = ec::AffinePoint::new_unchecked([0u32; 8], [0u32; 8]);
+    for (point, scalar) in xks_iter {
+        point.mul(&scalar, &mut mul_buffer);
+        // TODO experiment with alternating buffers for perf
+        result.add(&mul_buffer, &mut result_buffer);
+        core::mem::swap(&mut result, &mut result_buffer);
```

Comment:
Does this swap the actual values, or just the pointers? It might be worth defining `result_buffer` as type `&mut ec::AffinePoint` and then swapping the references here. Not 100% sure that will work, will produce better results, and that Rust will not complain.

Reply:
That's not possible, because you need to access the value immutable in-between where the buffers are defined and where this swap is.

I understand what you're saying, and will think more about this and resolve this TODO (or measure with my initial plan) before it gets merged.

Reply:
I was wrong, it is possible to just use `mem::swap` on the pointers, and it gives roughly the same perf as just alternating buffers, but much cleaner. Assume it's roughly the same as alternating because swapping the address in memory is likely as cheap as %2 checks (just a guess)

[`a12c63a` (#6)](https://github.com/risc0/RustCrypto-elliptic-curves/pull/6/commits/a12c63a066344d2990c93e5cc4e915019b90b6ab)